### DISCUSSION TITLE
Questions about Ray implementation of PPO

### DISCUSSION THREAD

**Discussion by cryoco** [hpcaitech/ColossalAI#3606] [Development | Core]
I've seen 2 ray implementations of PPO in this repo, [#3195 ](https://github.com/hpcaitech/ColossalAI/pull/3195) and [#3309 ](https://github.com/hpcaitech/ColossalAI/pull/3309). The former makes the experience maker and trainer decoupled, whereas the latter's models are still shared by experience maker and trainer. I'd like to know which implementation will be maintained in the future. Thank you!

**Comment by binmakeswell** [comment]
Hi @cryoco decoupled version. It is still under early development. There are many elements to be added and optimized.
